pkg/store/bucket.go

```diff
@@ -834,6 +825,62 @@ func (r *bucketIndexReader) preloadPostings() error {
 	return nil
 }
 
+// preloadPostings loads all given postings. It tries to avoid bigger gaps of irrelevant bytes.
+// TODO(bplotka): Merge this unit-tested function with the preloadFile (for chunks) and apply same for prealodSeries.
+func preloadPostings(
+	ctx context.Context,
+	postings []*lazyPostings,
+	loadPosting func(ctx context.Context, postings []*lazyPostings, start int64, length int64) error,
+) error {
+	if len(postings) == 0 {
+		return nil
+	}
+
+	sort.Slice(postings, func(i, j int) bool {
+		return postings[i].ptr.Start < postings[j].ptr.Start
+	})
+
+	// Maximum amount of irrelevant bytes between postings we are willing to fetch.
+	const maxPostingsGap = 1024
+
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
Why are we creating a new context here? It's cancel method is never called either.

Reply:
yup - it is used by run.Group below.